### ISSUE TITLE
Add local file reading for form/file request

### ISSUE THREAD

**Issue by SwapBuffers** [rangav/thunder-client-support#447]
**Is your feature request related to a problem? Please describe.**
I've been using the code-server version of vscode which allows me to code remotely in the browser. This plugin isn't available on the marketplace I assume because it's not fully supported. I went ahead and installed the plugin and it was great, good job. However I've run into an issue when submitting POST requests with files. The problem is files can't be uploaded through the browser this way.

**Describe the solution you'd like**
Could we have an option within the body/form tab under the Files to select between a local/remote file. The solution doesn't need to be all that complex, simplest would be in my opinion remove the readonly attribute on the file-path-value and during the submission of the form (submit request) it checks if a file is present and if no file is present it tries to read from the file-path-value that the user entered (local file path).

**Describe alternatives you've considered**
I have tried httpbook extension for vscode which lets me specify a local input file but doesn't have the nice GUI and collection/environment features ThunderClient offers.

**Implementation:**
Is this open source? I can't find source code to investigate further.

**Comment by rangav** [comment]
Hi @SwapBuffers thanks for the feedback, will add to roadmap.

**Comment by SwapBuffers** [comment]
Thanks @rangav in the interim could you shed any light on the format of the files[] array in the thunderclient.db, is it possible for me to set a file+array/blob in there to get me going manually? or is there any way to load a file into an environment variable and use that somehow?

Just clutching at straws to get me by.

**Comment by rangav** [comment]
- Create new collection and one new request with files
- Export the collection 
- Open the json file in vscode
- Manually edit files array paths
- Then re-import that json from collection tab Menu


See if it works

**Comment by rangav** [comment]
Please read this comment, not sure if it's related
https://github.com/rangav/thunder-client-support/issues/158#issuecomment-851058322

**Comment by SwapBuffers** [comment]
> * Create new collection and one new request with files
> * Export the collection
> * Open the json file in vscode
> * Manually edit files array paths
> * Then re-import that json from collection tab Menu
> 
> See if it works

Unfortunately this doesn't work (just get an empty files:[] ) I suspect it's due to how code-server integrates vscode. When I select a file it doesn't update the input-file-value to show the path either, it's as if it can't read the file. 

Would it be possible from anyone to get a copy/paste an of an exported collection with a form file so I can copy from that and see if that gets me by. Thanks in advance.

**Comment by rangav** [comment]
When you select file, it will only show name of file.

did you save request?

let me know how to install the extension on code-server,  so I can test and get back to you.

**Comment by SwapBuffers** [comment]
> When you select file, it will only show name of file.
> 
> did you save request?
> 
> let me know how to install the extension on code-server, so I can test and get back to you.

Setting the extensionKind to ui disables the plugin in code-server with a link mentioning more information to here:
https://code.visualstudio.com/api/advanced-topics/remote-extensions#architecture-and-extension-types

Installing your extension under code-server is just a manual install vsix CTRL+SHIFT+P ext i VSIX from file. I could then choose the extension.vsix and install it that way. Everything in there works fine so far just the file selection.

**Comment by SwapBuffers** [comment]
> When you select file, it will only show name of file.
> 
> did you save request?
> 
> let me know how to install the extension on code-server, so I can test and get back to you.

Saving the request isn't an option, though I can trigger save by pressing CTRL+S and then backing out of the browsers save page dialog. This issue sounds related to #372 however that covers just saving the request. Still have the issue of reading the file over code-server or even a local file. Hence why I want to try an import which references a local file on the code-server and see if that will get me by.

**Comment by rangav** [comment]
Just run the request it will save the request 

code-server is there free version for testing?

**Comment by SwapBuffers** [comment]
> Just run the request it will save the request
> 
> code-server is there free version for testing?

Yeah code-server is free, just run the install script and run the service. you'll need to access it over https, you can do that for free with cloudflare and nginx/apache with proxy. If it would help I could set up one and give you access?

**Comment by rangav** [comment]
if you can setup, that would be faster. Please share details when its ready. Thanks

**Comment by SwapBuffers** [comment]
> if you can setup, that would be faster. Please share details when its ready. Thanks

Can I get your public key from here?

**Comment by rangav** [comment]
what public key? use any random word

**Comment by rangav** [comment]
i am closing this issue as there is similar issue already exists related to this, please make any future comments there.

duplicate of #406
